packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -0,0 +1,254 @@
+/// [WidgetSpan]s. Child [Text] and [RichText] widgets will be laid out
+/// independently and occupy a rectangular space in the parent text layout.
+///
+/// [WidgetSpan]s will be ignored when passed into a [TextPainter] directly.
+/// To properly layout and paint the [widget], [WidgetSpan] should be passed
+/// into a [Text.rich] widget.
+///
+/// {@tool sample}
+///
+/// A card with `Hello World!` embedded inline within a TextSpan tree.
+///
+/// ```dart
+/// <TextSpan>[
+///   TextSpan(text: 'Flutter is'),
+///   WidgetSpan(
+///     widget: SizedBox(
+///       width: 120,
+///       height: 50,
+///       child: Card(
+///         child: Center(
+///           child: Text('Hello World!')
+///         )
+///       ),
+///     )
+///   ),
+///   TextSpan(text: 'the best!'),
+/// ]
+/// ```
+/// {@end-tool}
+///
+/// The [semanticsLabel] argument may be provided to supply a description of
+/// this [WidgetSpan] within the [toPlainText]. The semantics label of the
+/// overall [InlineSpan] tree will not mention [WidgetSpan]s unless
+/// [semanticsLabel] is provided. Semantics for the widget itself will still
+/// function independently from the [InlineSpan] semantics label.
+///
+/// See also:
+///
+///  * [TextSpan], a node that represents text in a [InlineSpan] tree.
+///  * [Text], a widget for showing uniformly-styled text.
+///  * [RichText], a widget for finer control of text rendering.
+///  * [TextPainter], a class for painting [InlineSpan] objects on a [Canvas].
+@immutable
+class WidgetSpan extends PlaceholderSpan {
+  /// Creates a [WidgetSpan] with the given values.
+  ///
+  /// The [child] property should be non-null. [WidgetSpan] is a leaf node in
+  /// the [InlineSpan] tree
+  ///
+  /// A [TextStyle] may be provided with the [style] property, but only the
+  /// decoration, foreground, background, and spacing options will be used.
+  const WidgetSpan({
+    @required this.child,
+    InlineWidgetAlignment alignment = InlineWidgetAlignment.bottom,
+    TextBaseline baseline,
+    TextStyle style,
+  }) : assert(child != null),
+       assert((alignment == InlineWidgetAlignment.aboveBaseline ||
+               alignment == InlineWidgetAlignment.belowBaseline ||
+               alignment == InlineWidgetAlignment.baseline) ? baseline != null : true),
+       super(alignment:
+           // Convert InlineWidgetAlignment to PlaceholderAlignment in a const fashion.
+           alignment == InlineWidgetAlignment.baseline ? ui.PlaceholderAlignment.baseline :
+           alignment == InlineWidgetAlignment.aboveBaseline ? ui.PlaceholderAlignment.aboveBaseline :
+           alignment == InlineWidgetAlignment.belowBaseline ? ui.PlaceholderAlignment.belowBaseline :
+           alignment == InlineWidgetAlignment.top ? ui.PlaceholderAlignment.top :
+           alignment == InlineWidgetAlignment.bottom ? ui.PlaceholderAlignment.bottom :
+           alignment == InlineWidgetAlignment.middle ? ui.PlaceholderAlignment.middle : null,
```

Comment:
if these are identical, why don't we just reexport PlaceholderAlignment and use that directly?